codec_packetizers/h265/h265_packetizer.c

```diff
@@ -0,0 +1,605 @@
+                if( memcmp( &( pFrame->pFrameData[currentIndex] ),
+                            &( startCode2[0] ),
+                            sizeof( startCode2 ) ) == 0 )
+                {
+                    if( firstStartCode == 1 )
+                    {
+                        firstStartCode = 0;
+                    }
+                    else
+                    {
+/* Create NAL unit from data between start codes */
+                        nalu.pNaluData = &( pFrame->pFrameData[naluStartIndex] );
+                        nalu.naluDataLength = currentIndex - naluStartIndex;
+
+/* Extract H.265 header fields if NAL unit is big enough */
+                        if( nalu.naluDataLength >= NALU_HEADER_SIZE )
+                        {
+                            nalu.nal_unit_type = ( nalu.pNaluData[0] >> 1 ) & 0x3F;
+                            nalu.nal_layer_id = ( ( nalu.pNaluData[0] & 0x01 ) << 5 ) |
+                                                ( ( nalu.pNaluData[1] >> 3 ) & 0x1F );
+                            nalu.temporal_id = nalu.pNaluData[1] & 0x07;
+
+                            result = H265Packetizer_AddNalu( pCtx,
+                                                             &nalu );
+                        }
+                        else
+                        {
+                            result = H265_RESULT_MALFORMED_PACKET;
+                        }
+                    }
+
+                    naluStartIndex = currentIndex + sizeof( startCode2 );
+                    currentIndex = naluStartIndex;
+                    continue;
+                }
+            }
+
+            currentIndex++;
+        }
+
+/* Handle last NAL unit in frame */
+        if( ( result == H265_RESULT_OK ) && ( naluStartIndex > 0 ) )
+        {
+            nalu.pNaluData = &( pFrame->pFrameData[naluStartIndex] );
+            nalu.naluDataLength = pFrame->frameDataLength - naluStartIndex;
+
+            if( nalu.naluDataLength >= NALU_HEADER_SIZE )
+            {
+                nalu.nal_unit_type = ( nalu.pNaluData[0] >> 1 ) & 0x3F;
+                nalu.nal_layer_id = ( ( nalu.pNaluData[0] & 0x01 ) << 5 ) |
+                                    ( ( nalu.pNaluData[1] >> 3 ) & 0x1F );
+                nalu.temporal_id = nalu.pNaluData[1] & 0x07;
+
+                result = H265Packetizer_AddNalu( pCtx,
+                                                 &nalu );
+            }
+            else
+            {
+                result = H265_RESULT_MALFORMED_PACKET;
+            }
+        }
+    }
+
+    return result;
+}
+
+/*-----------------------------------------------------------------------------------------------------*/
+
+H265Result_t H265Packetizer_AddNalu( H265PacketizerContext_t * pCtx,
+                                     H265Nalu_t * pNalu )
+{
+    H265Result_t result = H265_RESULT_OK;
+
+    if ((pCtx == NULL) || (pNalu == NULL) || 
+        (pNalu->pNaluData == NULL) || (pNalu->naluDataLength == 0))
```

Comment:
```suggestion
    if( ( pCtx == NULL ) || ( pNalu == NULL ) ||
        ( pNalu->pNaluData == NULL ) || ( pNalu->naluDataLength == 0 ) )
```
Please follow the coding style in this repo.